docs/fundamentals/executor/executor-in-flow.md

```diff
@@ -1,136 +1,125 @@
+This example shows how to start a Flow with an Executor via the Python API:
 
-The list of the `runtime_args` is:
+```python
+with Flow().add(
+    uses='MyExecutor',
+    uses_with={
+      "parameter_1": "foo",
+      "parameter_2": "bar"
+    },
+    uses_metas={
+      "name": "MyExecutor",
+      "description": "MyExecutor does a thing to the stuff in your Documents",
+      "py_modules": ["executor.py"]
+    },
+    uses_requests={
+      "/index": "my_index",
+      "/search": "my_search",
+      "/random": "foo"
+    },
+    workspace="some_custom_path"
+) as f:
+  ...
+```
 
-- `name`: Name given to the `Executor`. This is dynamically adapted from the `name` in `metas` and depends on some additional arguments like `shard_id`. 
-- `replicas`: Number of replicas of the same `Executor` deployed with the `Flow`.
-- `shards`: Number of shards of the same `Executor` deployed with the `Flow`.
-- `shard_id`: Identifier of the `shard` corresponding to the given `Executor` instance.
-- `workspace`: Path to be used by the `Executor`. Note that the actual workspace directory used by the Executor is obtained by appending `'/<executor_name>/<shard_id>/'` to this value.
-- `py_modules`: Path to the modules needed to import the `Executor`. This is another way to pass `py-modules` to the `Executor` from the `Flow`
+Python API-specific options:
 
+- `uses` can be a class or string. Defines the Executor to load. You can also run Executors available on [Jina Hub](https://hub.jina.ai) with a {ref}`special syntax <use-hub-executor>`.
 
-````{admonition} Note
-:class: note
-The YAML API will ignore `.runtime_args` during save and load as they are not statically stored
 ````
 
-### Requests
-By default, an `Executor` object contains `.requests` as an attribute when loaded from the `Flow`. This attribute is a `Dict` describing the mapping between Executor methods and network endpoints: It holds endpoint strings as keys, and pointers to `function`s as values. 
-
-### Workspace
-Each `Executor` has a special *workspace* that is reserved for that specific Executor instance.
-The `.workspace` property contains the path to this workspace.
-
-This `workspace` is generated using the workspace passed when adding the Executor: `flow.add(..., workspace='path/to/workspace/')`.
-The value that is passed there is added to the `runtime_args`, and the final `workspace` is generated by appending `'/<executor_name>/<shard_id>/'`,
-as described above.
+````{tab} via YAML
 
-If the user hasn't provided a workspace, the Executor uses a default workspace, which is defined in the `JINA_DEFAULT_WORKSPACE_BASE`
-environment variable.
+```yaml
+  jtype: MyExecutor
+  uses_with:
```

Comment:
it would be interesting to see here how u can pass some `with` parameters directly in the `MyExecutor`

Reply:
We wanted to keep it consistent across examples, using `uses_X` in both. In Python you can't use the short form `with`